### DISCUSSION TITLE
[Question] configuration and performance of network coordinator

### DISCUSSION THREAD

**Discussion by groovepack** [Koenkk/zigbee2mqtt#14552] [Devices]
Hi,

first of all - many thanks to the team for your effort. I highly appreciate it!

So, ich have to change my old CC2531 based adapter as i have trouble getting a stable connection to all router devices (e.g. Hue bulbs) one or two floors higher. My home server, which runs node-red and zigbee2mqtt is located at the basement but alle routers are on higher floors.

Now i thought it would be better to place a network based coordinator on the middle floor. How is the performance compared to a usb connected coordinator. Is there a remarkable difference, e.g. latency?

Option 1) SONOFF Zigbee 3.0 USB Dongle Plus ZBDongle-P on basement
Option 2) CC2652P2 ZigBee Ethernet Coordinator on middle floor

And how is it configured in configuration.yaml ? The documentation only shows configuration of a serial-device on the same host.

Thank you in advance for your information und support!

**Comment by desertblade** [comment]
You could add Zigbee repeaters on your second floor. Using repeaters will create multiple paths to each endpoint and help create a robust mesh.  Often Zigbee smart plugs/outlet can be found for around $10 US and they almost always act as a repeater.  

For reference I have a 3 story house, and my zigbee usb stick is in the top corner. Run at least 1 repeater per floor, often 1 per primary room. Over the years it has grown to 6 in my house, and have another 70+ zigbee devices throughout.

You can use the MAP tab in Zigbee2MQTT to see the routes that have been created.
